lib/srv/sess_test.go

```diff
@@ -267,13 +267,32 @@ func TestInteractiveSession(t *testing.T) {
 
 	t.Run("BrokenRecorder", func(t *testing.T) {
 		t.Parallel()
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+
 		sess := testOpenSession(t, reg)
+		stateUpdate := sess.tracker.SubscribeToStateUpdate(ctx)
+
+	Outer:
+		for {
+			select {
+			case state := <-stateUpdate:
+				if state == types.SessionState_SessionStateRunning {
+					break Outer
+				}
+			case <-time.After(time.Second * 5):
+				t.Fatal("timeout waiting for session to start")
+			}
+		}
 
 		// The recorder might be closed in the case of an error downstream.
 		// Closing the session recorder should result in the session ending.
 		err := sess.recorder.Close(context.Background())
 		require.NoError(t, err)
-		require.Eventually(t, sess.isStopped, time.Second*5, time.Millisecond*500)
+
+		_, err = sess.inWriter.Write([]byte("foo"))
+		require.NoError(t, err)
```

Comment:
So we expect that `Write` will always succeed even though we've closed the recorder? Will this be suspect to timing?

Reply:
Write here will always succeed yes. The session doesn't yet know the recorder is closed. We're writing to the input pipe here which is seperately read and multiplexed onto the writers, at which point we notice the recorder is broken.